File: src/lidar_pkg/scripts/lidar_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from sensor_msgs.msg import LaserScan
from geometry_msgs.msg import Twist

def LaserScanCallBack(msg):
    dist = msg.ranges[180]
    rospy.loginfo("前方测距 ranges[180] = %f m",dist)
    vel_msg = Twist()
    if dist < 1.5:
        vel_msg.angular.z = 0.7
    else:
        vel_msg.linear.x = 0.05
    vel_pub.publish(vel_msg)

if __name__ == "__main__":
    rospy.init_node("lidar_node")
    lid_sub = rospy.Subscriber("/scan",LaserScan,LaserScanCallBack,queue_size=10)
    vel_pub = rospy.Publisher("/cmd_vel",Twist,queue_size=10)
    rospy.spin()

File: src/lidar_pkg_c/src/lidar_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>
#include <geometry_msgs/Twist.h>

geometry_msgs::Twist vel_msg;
ros::Publisher vel_pub;

void LidarCallBack(const sensor_msgs::LaserScan msg)
{
    float fMidDist = msg.ranges[180];
    ROS_INFO("Front distance: %f",fMidDist);
    if(fMidDist<2)
    {
        vel_msg.angular.z = 1;
    }
    else
    {
        vel_msg.angular.z = 0;
    }
    vel_pub.publish(vel_msg);
}
int main(int argc, char *argv[])
{
    ros::init(argc,argv,"lidar_node");
    ros::NodeHandle n;
    ros::Subscriber lidar_sub = n.subscribe("/scan",10,&LidarCallBack);
    vel_pub = n.advertise<geometry_msgs::Twist>("/cmd_vel",10);
    vel_msg.linear.x = 0.1;
    vel_msg.linear.y = 0;
    vel_msg.linear.z = 0;
    vel_msg.angular.x = 0;
    vel_msg.angular.y = 0;
    vel_msg.angular.z = 0;
    ros::spin();
    return 0;
}
